### PR TITLE
[RFC] mavlink: introduce the mavlink protocol implementation

### PR DESCRIPTION
**Pull request by dorileo**
## Known issues
  + This is not yet a full complaint implementation, there'are lots of API's yet to come;
  + The serial connection mechanism is not yet implemented;

## Rationale
This patch introduce the basic infrasctructure and API's to control MAV
(Micro Air Vehicle) communication. More API's are to come.

Signed-off-by: Leandro Dorileo <leandro.maciel.dorileo@intel.com>

### CODE REVIEW

File: src/lib/comms/sol-mavlink.c [hunk 1]
```diff
@@ -0,0 +1,786 @@
...
+    mavlink_message_t msg;
+    mavlink_status_t status;
+    uint8_t buf[MAVLINK_MAX_PACKET_LEN];
+    int i, res;
+    sol_mavlink_status sol_status = mavlink->status;
+
+    bzero(buf, MAVLINK_MAX_PACKET_LEN);
+    res = recv(mavlink->fd, buf, MAVLINK_MAX_PACKET_LEN, 0);
+
+    for (i = 0; i < res; ++i) {
```
Comment (lpereira) [inline src/lib/comms/sol-mavlink.c]: What happens if `recv()` returns `-1`?

File: src/lib/comms/sol-mavlink.c [hunk 2]
```diff
@@ -0,0 +1,786 @@
...
+sol_mavlink_fd_handler(void *data, int fd, uint32_t cond)
+{
+    struct sol_mavlink *mavlink = data;
+    mavlink_message_t msg;
+    mavlink_status_t status;
+    uint8_t buf[MAVLINK_MAX_PACKET_LEN];
+    int i, res;
+    sol_mavlink_status sol_status = mavlink->status;
+
+    bzero(buf, MAVLINK_MAX_PACKET_LEN);
```
Comment (lpereira) [inline src/lib/comms/sol-mavlink.c]: Maybe `uint8_t buf[...] = {0}` instead?

File: src/lib/comms/sol-mavlink.c (hunk 1 above)
Comment (dorileo) [inline src/lib/comms/sol-mavlink.c]: i will be greater than res and will not run the loop, nothing will take effect and so on, but well, worth testing and returning...

File: src/lib/comms/sol-mavlink.c (hunk 2 above)
Comment (dorileo) [inline src/lib/comms/sol-mavlink.c]: yep sure, the use of bzero and friends have been discussed by the other guys, I'm getting rid of it wherever appropriate.

**Comment by dorileo** [comment]
new version coming...

**Pull request closed without merging**